addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiClientThingHandler.java

```diff
@@ -0,0 +1,299 @@
+                    break;
+
+                // :site
+                case CHANNEL_SITE:
+                    if (clientHome && site != null && StringUtils.isNotBlank(site.getDescription())) {
+                        state = StringType.valueOf(site.getDescription());
+                    }
+                    break;
+
+                // :macAddress
+                case CHANNEL_MAC_ADDRESS:
+                    if (clientHome && client != null && StringUtils.isNotBlank(client.getMac())) {
+                        state = StringType.valueOf(client.getMac());
+                    }
+                    break;
+
+                // :ipAddress
+                case CHANNEL_IP_ADDRESS:
+                    if (clientHome && client != null && StringUtils.isNotBlank(client.getIp())) {
+                        state = StringType.valueOf(client.getIp());
+                    }
+                    break;
+
+                // :uptime
+                case CHANNEL_UPTIME:
+                    if (clientHome && client != null && client.getUptime() != null) {
+                        state = new DecimalType(client.getUptime());
+                    }
+                    break;
+
+                // :lastSeen
+                case CHANNEL_LAST_SEEN:
+                    // mgb: we don't check clientOnline as lastSeen is also included in the Insights data
+                    if (client != null && client.getLastSeen() != null) {
+                        state = new DateTimeType(client.getLastSeen());
+                    }
+                    break;
+
+                default:
+                    if (client != null) {
+                        // mgb: additional wired client channels
+                        if (client.isWired() && (client instanceof UniFiWiredClient)) {
+                            state = getWiredChannelState((UniFiWiredClient) client, clientHome, channelID);
+                        }
+
+                        // mgb: additional wireless client channels
+                        else if (client.isWireless() && (client instanceof UniFiWirelessClient)) {
+                            state = getWirelessChannelState((UniFiWirelessClient) client, clientHome, channelID);
+                        }
+                    }
+                    break;
+            }
+
+            // mgb: only non null states get updates
+            if (state != null) {
+                updateState(channelID, state);
+            }
+        }
+    }
+
+    private State getWiredChannelState(UniFiWiredClient client, boolean clientHome, String channelID) {
+        State state = null;
+        return state;
+    }
+
+    private State getWirelessChannelState(UniFiWirelessClient client, boolean clientHome, String channelID) {
+        State state = null;
+        switch (channelID) {
+            // :ap
+            case CHANNEL_AP:
+                UniFiDevice device = client.getDevice();
```

Comment:
In the other switch cases `client` is checked for `null`, but not here. (You could add `@NonNullByDefault` to this class to help with this. Although this might be a version 2 feature as it requires some more work). But the null check might be needed here.

Reply:
These two `getWirelessChannelState` and `getWiredChannelState` had a null check before they're called from 

https://github.com/mgbowman/openhab2-addons/blob/unifi-binding/addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/handler/UniFiClientThingHandler.java#L248-L259

I went ahead and added `@NonNull` to the parameters and removed the existing null checks.

Reply:
It's the correct way now with `@NonNullByDefault` at the class level